src/model/differ.js

```diff
@@ -0,0 +1,703 @@
+
+					j++;
+				} else if ( action == 'charAttr' ) {
+					// A character had its attribute changed.
+					const range = Range.createFromParentsAndOffsets( element, i, element, i + 1 );
+
+					// Generate diff items (one for each attribute) for this character and insert them into the diff set.
+					diffSet.push( ...this._getAttributesDiff( range, snapshotChildren[ j ].attributes, elementChildren[ i ].attributes ) );
+
+					i++;
+					j++;
+				} else {
+					// `action` is 'equal'. Child not changed.
+					i++;
+					j++;
+				}
+			}
+		}
+
+		// After changes in child nodes are handled, process elements that changed.
+		for ( const [ element, snapshot ] of this._changedElements ) {
+			if ( element.name == snapshot.name ) {
+				// Name is the same, so only attributes of that element changed.
+				const range = Range.createFromParentsAndOffsets( element.parent, element.startOffset, element, 0 );
+
+				// Generate diff items (one for each attribute) for this element and insert them into the diff set.
+				diffSet.push( ...this._getAttributesDiff( range, snapshot.attributes, element.getAttributes() ) );
+			} else {
+				// Element name has changed. Since this requires removing old element and inserting new element,
+				// we don't care about attributes. If those have changed, they will be refreshed with the newly inserted element.
+				// Generate diff items for element "refreshing".
+				diffSet.push( this._getRemoveDiff( element.parent, element.startOffset, snapshot.name ) );
+				diffSet.push( this._getInsertDiff( element.parent, element.startOffset, element.name ) );
+			}
+		}
+
+		// At this point, all buffered changes has been processed and the diff set is calculated. Now, it will be further processed.
+		// First, remove all changes that are included in other changes.
+		for ( let i = 0; i < diffSet.length; i++ ) {
+			for ( let j = 0; j < diffSet.length; j++ ) {
+				if ( i !== j ) {
+
+					if ( diffItemIncludes( diffSet[ i ], diffSet[ j ] ) ) {
+						diffSet.splice( j, 1 );
+						j--;
+						break;
+					}
+				}
+			}
+		}
+
+		// Then, sort the changes by the position (change at position before other changes is first).
+		diffSet.sort( ( a, b ) => {
+			// If the change is in different root, we don't care much, but we'd like to have all changes in given
+			// root "together" in the array. So let's just sort them by the root name. It does not matter which root
+			// will be processed first.
+			if ( a.position.root != b.position.root ) {
+				return a.position.root.rootName < b.position.root.rootName ? -1 : 1;
+			}
+
+			// If change happens at the same position...
+			if ( a.position.isEqual( b.position ) ) {
+				// Keep chronological order of operations.
+				return a.changeCount < b.changeCount ? -1 : 1;
+			}
+
+			// If positions differ, position "on the left" should be earlier in the result.
+			return a.position.isBefore( b.position ) ? -1 : 1;
+		} );
+
+		// Glue together multiple changes (mostly on text nodes).
```

Comment:
It would be nice to handle text with multiple changed attributes. It can be done as a follow-up.